source/slides/05_linux_basics.rst

```diff
@@ -0,0 +1,366 @@
+  :align: center
+  :width: 75%
+
+LVM commands
+------------
+
+.. code-block:: bash
+
+  $ yum install lvm2
+
+  # Note: I created loop1/2 using dd and losetup
+  $ pvcreate /dev/loop1 /dev/loop2
+    Physical volume "/dev/loop1" successfully created
+    Physical volume "/dev/loop2" successfully created
+
+  $ vgcreate lvm /dev/loop1 /dev/loop2
+    Volume group "lvm" successfully created
+
+  $ lvcreate -n home -L 1G lvm
+    Logical volume "home" created
+
+  $ mkfs.ext4 /dev/lvm/home
+  $ mount /dev/lvm/home /mnt
+
+LVM commands
+------------
+
+All commands start with ``pv``, ``vg`` or ``lv``
+
+::
+
+  pvchange   pvcreate   pvmove     pvresize   pvscan
+  pvck       pvdisplay  pvremove   pvs
+
+  vgcfgbackup    vgconvert      vgextend       vgmknodes      vgs
+  vgcfgrestore   vgcreate       vgimport       vgreduce       vgscan
+  vgchange       vgdisplay      vgimportclone  vgremove       vgsplit
+  vgck           vgexport       vgmerge        vgrename
+
+  lvchange     lvextend     lvmdiskscan  lvmsar       lvresize
+  lvconvert    lvm          lvmdump      lvreduce     lvs
+  lvcreate     lvmchange    lvmetad      lvremove     lvscan
+  lvdisplay    lvmconf      lvmsadc      lvrename
+
+Useful tricks with LVM
+----------------------
+
+* Move LE's between PV's (linear vs. striped problem)
+* Move from one PV to another PV live
+* Rename VG's or LV's live
+* Reduce, remove or add LV's as needed
+* Make atomic snapshots for backups
+
+Issues with LVM
+---------------
+
+* Requires an initrd if the rootfs is in lvm
+
+  * Can be make troubleshooting difficult sometimes
+
+* Can't move LV's between VG's
+* If a disk dies, all the PE's are gone (unless RAID is used in LVM)
+* Using LVM inside of LVM can be hard to troubleshoot
+
+  * Ganeti VMs
+
+RAID
+-----
+
+**Software (mdadm or LVM)**
+
+* sda1/sda2 -> md0 -> PV -> VG -> LV - > filesystem
+* sda1/sda2 -> PV -> VG -> -LV (raid#) -> filesystem
+* mdadm + lvm usually preferred
+* Performance has improved quite a bit over the years
+
+**Hardware**
+
+* Dedicated hardware to RAID
+* Prevent boot problems that mdadm may have
+* Proprietary tools
+* Claim better performance
+
+RAID Levels
+-----------
+
+.. csv-table::
+  :header: Level, Minimum disks, Description
+
+  RAID0, 2, Striping
+  RAID1, 2, Mirroring
+  RAID5, 3, Block-level striping with distributed parity
+  RAID6, 4, Block-level striping with double distributed parity
+  RAID10/RAID1+0, 4, Striped set from a series of mirrored drives
+
+*Hot spares can automate rebuilds (i.e. RAID5+1 & RAID6+1)*
+
+RAID Issues
+-----------
+
+* Complicate data recovery
+* HW Controller limitations and cost
+* HW management utility differences
+* Multiple disk failures during rebuild
+* Performance
+
+Automate OS Installation
+========================
+
+Ideal Requirements
+------------------
+
+*These are not entirely required, but makes it a lot easier*
+
+* Working DHCP environment
+* Network booting capable hardware (i.e. PXE)
```

Comment:
Might be worth mentioning iPXE vs PXE somewhere, because ipxe allows you to do some wicked cool things, like [pull the image over http instead of tftp](http://staff.osuosl.org/~jordane/coreos.ipxe). It also allows pxe over wifi, which is sweet for laptops.